lib/turn.rb

```diff
@@ -1,12 +1,72 @@
 class Turn
-  attr_accessor :board, :player, :turn_number
+  attr_accessor :board, :player, :computer, :turn_number
+  
   def initialize(board, player)
     @board = board
     @player = player
-    # @computer = computer
+    @computer = computer
     @turn_number = 0
+  end
+
+  def place_selection
+    p "Choose a column A-G to drop your piece!"
+    selection = gets.chomp.upcase
 
+    until @board.grid.first.include?(selection)
+      p "Please input a column letter A-G."
+      selection = gets.chomp.upcase
+    end
+    column_converter(selection)
   end
 
 
-end
+  def column_converter(selection)
+    # selection_letters = @board.grid[0]
+    
+    # selection_letters.each_with_index do |l, index|
+    #   index_column = index if selection == l
+    # end
+    if selection == "A"
+      index_column = 0 
+    elsif selection == "B"
+      index_column = 1
+    elsif selection == "C"
+      index_column = 2
+    elsif selection == "D"
+      index_column = 3
+    elsif selection == "E"
+      index_column = 4
+    elsif selection == "F"
+      index_column = 5
+    elsif selection == "G"
+      index_column = 6
+    end
+  end 
+  
+
+  def column_avail?(index_column)
```

Comment:
column_avail? method added to show available slots. If the slot is ".", it returns true(available), if the index row is less than 1, no more available slots or no ".", will return false. Starting from the bottom row(>=1)  to the top row. (-=1)